os/wqueue/kwqueue/kwork_signal.c

```diff
@@ -88,5 +91,21 @@ int work_signal(int qid)
 		return -EINVAL;
 	}
 
-	return work_qsignal(pid);
+	/* Signal must be sent only when worker thread is sigwait state.
+	 * Otherwise, unnecessary signal will be sent(when worker thread is
+	 * processing the work queue).
+	 */
+
+	state = task_getstate(pid);
+	if (state == TSTATE_WAIT_SIG) {
+		/* Send signal to wake up the signal waiting thread */
+		return work_qsignal(pid);
+	} else if (state >= 0) {
+		/* It's already working, just return without signaling */
+		return OK;
+	} else {
+		/* Less than zero, No task with this pid was found */
+		sdbg("ERROR! LPWROK pid = %d is not alive\n", pid);
```

Comment:
Is it necessarily LPWORK pid or could be HPWORK as well ?
May be needs to be checked as in line 59

Reply:
oh, Thank you I checked this.